stdlib/asm/crypto/stark/deep_queries.masm

```diff
@@ -467,11 +459,34 @@ end
 #!
 #! Input: [query_ptr, ...]
 #! Output: [...]
-#! Cycles: 6 + num_queries * 473
+#! Cycles: 24 + num_queries * 445
 export.compute_deep_composition_polynomial_queries
+
     exec.constants::fri_com_ptr
     dup.1
-    #=>[query_ptr, query_end_ptr, ...]
+    push.0
+    exec.constants::deep_rand_coef_ptr
+    exec.constants::ood_trace_ptr
+    exec.constants::current_trace_row_ptr
+    exec.constants::tmp9 mem_storew
+    #=>[X, query_ptr, query_end_ptr, ...]
+
+    # Compute the negations of z and gz.
+    # We do it here as this computation is common to all queries.
+    exec.constants::z_ptr mem_loadw
```

Comment:
Could we add the description of the stack state after the `mem_loadw`?

Reply:
Added